File: examples/amplitude/app/src/main/java/com/trackingplan/examples/amplitude/MainActivity.java
package com.trackingplan.examples.amplitude;

import android.os.Bundle;

import com.google.android.material.snackbar.Snackbar;

import androidx.appcompat.app.AppCompatActivity;

import android.view.View;

import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import com.trackingplan.client.sdk.Trackingplan;
import com.trackingplan.examples.amplitude.databinding.ActivityMainBinding;

import android.view.Menu;
import android.view.MenuItem;

import com.amplitude.api.Amplitude;
import com.amplitude.api.AmplitudeClient;

import org.json.JSONException;
import org.json.JSONObject;

public class MainActivity extends AppCompatActivity {

    private AppBarConfiguration appBarConfiguration;
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Trackingplan.init("YOUR_TP_ID")
                .enableDebug()
                .dryRun()
                .start(this);

        AmplitudeClient client = Amplitude.getInstance()
                .initialize(getApplicationContext(), "ba5f6f06cfe2562435cc7eb7fcef28db")
                .enableForegroundTracking(getApplication());

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        setSupportActionBar(binding.toolbar);

        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment_content_main);
        appBarConfiguration = new AppBarConfiguration.Builder(navController.getGraph()).build();
        NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration);

        binding.fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                JSONObject eventProperties = new JSONObject();

                try {
                    eventProperties.put("Hover Time", 10).put("prop_2", "value_2");
                } catch (JSONException e) {
                    System.err.println("Invalid JSON");
                    e.printStackTrace();
                }

                client.logEvent("Button Clicked", eventProperties);
                client.uploadEvents();

                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                        .setAction("Action", null).show();
            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    public boolean onSupportNavigateUp() {
        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment_content_main);
        return NavigationUI.navigateUp(navController, appBarConfiguration)
                || super.onSupportNavigateUp();
    }
}

File: examples/customerio/app/src/main/java/com/trackingplan/examples/customerioexample/MainActivity.kt
package com.trackingplan.examples.customerioexample

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.trackingplan.client.sdk.Trackingplan
import com.trackingplan.examples.customerioexample.databinding.ActivityMainBinding
import io.customer.sdk.CustomerIO
import io.customer.sdk.data.model.Region

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        Trackingplan.init("YOUR_TP_ID")
            // .environment("PRODUCTION")
            // .sourceAlias("Android Example")
            // .customDomains(customDomains)
            .enableDebug()
            .dryRun()
            .start(this)

        CustomerIO.Builder(
            siteId = BuildConfig.CUSTOMER_IO_SITE_ID,
            apiKey = BuildConfig.CUSTOMER_IO_API_KEY,
            appContext = this.application)
            // .setRegion(Region.EU)
            .autoTrackScreenViews(true)
            .build()

        CustomerIO.instance()
            .identify(
                identifier = "TEST1",
                attributes = mapOf("first_name" to "Nick", "email" to "test@example.com")
            )


        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val navView: BottomNavigationView = binding.navView

        val navController = findNavController(R.id.nav_host_fragment_activity_main)
        // Passing each menu ID as a set of Ids because each
        // menu should be considered as top level destinations.
        val appBarConfiguration = AppBarConfiguration(
            setOf(
                R.id.navigation_home, R.id.navigation_dashboard, R.id.navigation_notifications
            )
        )
        setupActionBarWithNavController(navController, appBarConfiguration)
        navView.setupWithNavController(navController)
    }
}